EIPS/eip-6780.md

```diff
@@ -0,0 +1,65 @@
+
+ - Does not delete any storage keys or code
+ - Transfer the account balance to the target **and** set account balance to `0.`
+ - Note that no refund is given since [EIP-3529](./eip-3529.md).
+ - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+  
+2. Case where `SELFDESTRUCT` is executed in the same transaction the contract was created
+
+ - `SELFDESTRUCT` continues to behave as originally, i.e. deletes all storage keys and the account itself.
+ - Subsequently, the account will behave like exactly like an empty account, both in the same transaction and in all later ones
+ - Transfer the account balance to the target **and** set account balance to `0.`
+ - Note that no refund is given since [EIP-3529](./eip-3529.md).
+ - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+ - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+
+The `SELFDESTRUCT` opcode remains deprecated as specified in [EIP-6049](./eip-6049.md). Any use in newly deployed contracts is strongly discouraged even if this new behaviour is taken into account, and future changes to the EVM might further reduce the functionality of the opcode.
+
+## Rationale
+
+Getting rid of the `SELFDESTRUCT` opcode has been considered in the past, and there are currently no strong reasons to use it. Disabling it will be a requirement for statelessness. This EIP implements a behaviour that will attempt to leave some known uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning (as suggested in [EIP-6190](./eip-6190.md)). A further option that was considered was to just remove storage clearing from `SELFDESTRUCT` in [EIP-6046](./eip-6046.md) while preserving removal of contracts, however this is not safe because existing contracts rely on storage being empty when they are deployed.
+
+## Backwards Compatibility
+
+This EIP requires a hard fork, since it modifies consensus rules.
+
+Few applications are affected by this change. The only use that breaks is where a contract is re-created at the same address using `CREATE2` (after a `SELFDESTRUCT`), and the `SELFDESTRUCT` is executed in a transaction that's different from the one that creates the contract.
+
+## Security Considerations
+
+The following applications of `SELFDESTRUCT` will be broken and applications that use it in this way are not safe anymore:
+
+1. Any use where `SELFDESTRUCT` is used to burn non-ETH token balances, such as [ERC-20](./eip-20.md)), inside a contract. We do not know of any such use (since it can easily be done by sending to a burn address this seems an unlikely way to use `SELFDESTRUCT`)
```

Comment:
```suggestion
```
This is not a good use case of selfdestruct, with no known deployments, so this hardly seems worth mentioning (as you seem to indicate, yourself)